### DISCUSSION TITLE
Migration from 4 to 5 (or 6)?

### DISCUSSION THREAD

**Discussion by frontyard** [keystonejs/keystone#8218] [Questions]
We have a small project that has been running in production on Keystone 4 for many years.

We are at the point where things are starting to not "work" any more, for example we are locked on an old version of mongodb because mongoose version Keystone 4 is using doesn't support newer versions of mongodb. It's just a question of time when we'll no longer be able to update any part of our solution. Needless to say it's not an ideal situation.

Upgrading from 4 to 5 has been a terrifying prospect for us as we are very (very) small team and there isn't exactly a straight migration path. Now with version 6 we see that even mongodb is no longer supported. So not only we would need to rewrite our app, we would need to change our database and somehow migrate production data from nosql to sql solution.

Although we are very grateful for Keystone, it's hard not to be upset with some decision team has made over the years that left us in a difficult position. And while on an intelectual level I can see the reasoning behind it, on a practical level there doesn't seem to be a way out. And I'm sure we're not the only ones.

Does anyone have any reasonable advice on what to do?

**Comment by pcraciunoiu** [comment]
@frontyard did you ever find a good path here? I just started work with a client who is in a similar situation on Keystone 4, and are considering upgrade to 5 and then 6, with the challenge of dropping mongodb as well. Curious what steps you took and if you found any guides to upgrade 4 -> 5 and 5 -> 6.

**Reply by frontyard** [reply]
@pcraciunoiu No we haven't found a path forward. Even if we embarked on rewriting our app in v6, moving from mongodb to some sql db is not a reasonable burden to put on projects using your library. So we will most likely drop keystone, have a huge new project to rewrite our solultion without keystone and move on. Really sad.

**Reply by pcraciunoiu** [reply]
@frontyard I appreciate the follow-up and that's a shame you haven't found a path yet. I'm evaluating options for a current client and this is making me consider a rewrite as well. If you'd like to connect and share experiences you can also reach me through https://www.uplift.ltd/ - we often work with founders and small companies who may not have software tech in house to help set them up for success, and I'd be happy to help or learn from your experience. Best of luck either way!

**Reply by crowncode-development** [reply]
I've just found [this discussion](https://github.com/keystonejs/keystone/discussions/8490) but it about list of challenges not a solutions.
